central/graphql/resolvers/images_test.go

```diff
@@ -217,22 +265,57 @@ func (s *ImageResolversTestSuite) TestDeployments() {
 
 				// Test ImageCVECountBySeverity for each deployment resolver.
 				expectedCVESevCount := compileExpectedCountBySeverity([]*storage.Image{expectedImages[imageID]}, tc.vulnFilter)
-				actualCVECnt, err := image.ImageCVECountBySeverity(ctx, RawQuery{Query: paginatedQ.Query})
+				actualCVECnt, err := image.ImageCVECountBySeverity(testCtx, RawQuery{Query: paginatedQ.Query})
 				assert.NoError(t, err)
 
-				critical, err := actualCVECnt.Critical(ctx)
+				critical, err := actualCVECnt.Critical(testCtx)
 				assert.NoError(t, err)
-				important, err := actualCVECnt.Important(ctx)
+				important, err := actualCVECnt.Important(testCtx)
 				assert.NoError(t, err)
-				moderate, err := actualCVECnt.Moderate(ctx)
+				moderate, err := actualCVECnt.Moderate(testCtx)
 				assert.NoError(t, err)
-				low, err := actualCVECnt.Low(ctx)
+				low, err := actualCVECnt.Low(testCtx)
+				assert.NoError(t, err)
+
+				assert.Equal(t, int32(expectedCVESevCount.critical), critical.Total(testCtx))
+				assert.Equal(t, int32(expectedCVESevCount.important), important.Total(testCtx))
+				assert.Equal(t, int32(expectedCVESevCount.moderate), moderate.Total(testCtx))
+				assert.Equal(t, int32(expectedCVESevCount.low), low.Total(testCtx))
+
+				// Test image -> deployments -> images
+				imageDeployments, err := image.Deployments(testCtx, paginatedQ)
 				assert.NoError(t, err)
+				for _, d := range imageDeployments {
+					deploymentFound := false
+					deploymentID := d.Id(testCtx)
+					for _, expectedDeployment := range expectedDeployments {
```

Comment:
We could build an array of IDs and then assert equality of elements. Currently the test only asserts subsets.